### PR TITLE
merge in new scaling factor

### PR DESCRIPTION
**Pull request by ivica3730k**

### CODE REVIEW

**Comment by ivica3730k** [comment]
@james-hinton if you dont mind checking this out with me

File: workflows/utils/prepS2.py [hunk 1]
```diff
@@ -326,6 +328,73 @@ def s2_ndvi(red_file, nir_file, out_file=False):
...
+        img_arr_rescaled = img_arr_rescaled.astype(new_dtype)
+
+        # Making sure any original nodata is still nodata (I think?)
+        img_arr_rescaled.data[mask] = nodata
+
+        logging.info(f"MIN, MAX AFTER {img_arr_rescaled.min()}, {img_arr_rescaled.max()}")
+        logging.info(f"Data-type AFTER: {img_arr_rescaled.dtype}")
+
+        # Convert xarray to COG 
+        img_arr_rescaled.rio.to_raster(raster_path = out_path, driver="COG")
```
Comment (james-hinton) [inline workflows/utils/prepS2.py:396]: does this retain CRS and other metadata?

File: workflows/utils/dc_utilities.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-import gdal, osr
+from osgeo import gdal, osr
```
Comment (ivica3730k) [inline workflows/utils/dc_utilities.py:22]: was osr not needed?

File: workflows/utils/dc_water_classifier.py [hunk 3]
```diff
@@ -32,7 +32,7 @@
 import argparse
 import os
 import collections
-import gdal
+from osgeo import gdal
```
Comment (ivica3730k) [inline workflows/utils/dc_water_classifier.py:35]: did this import not work?

File: workflows/utils/prepLS.py [hunk 4]
```diff
@@ -8,6 +7,8 @@
 import tarfile
 from workflows.utils.prep_utils import *
 from typing import List
+import shutil
+import rioxarray as rxr
```
Comment (ivica3730k) [inline workflows/utils/prepLS.py:11]: check if rioxarray is installed

Comment (ivica3730k) [reply workflows/utils/prepLS.py:11]: it is

**Comment by ivica3730k** [comment]
LGTM (lets gamble, try merging)

File: workflows/utils/prepS2.py (hunk 1 above)
Comment (ivica3730k) [reply workflows/utils/prepS2.py:396]: uhh havent checked, @sabinenix ?

Comment (sabinenix) [reply workflows/utils/prepS2.py:396]: It looks to me like the outputs are still located correctly relative to the originals and I think the metadata itself is copied from the original download directory so nothing should be lost there. With the landsat data though they get re-cogged after scaling using the original code's cog function - do you think that order would that be preferable for sentinel too?

**Comment by sabinenix** [comment]
> LGTM (lets gamble, try merging) While I was testing I had the parts that uploaded to S3 all commented out, so just noticed that the directories might need to be changed there to point to where the scaled cogs are stored before that part of it works properly!

File: workflows/utils/prepS2.py (hunk 1 above)
Comment (james-hinton) [reply workflows/utils/prepS2.py:396]: I think that re-cogging is a good idea for sentinel

**Pull request merged**